modules/snooker/src/main/ts/ephox/snooker/model/DetailsList.ts

```diff
@@ -9,38 +9,55 @@ import { getAttrValue } from '../util/CellUtils';
    element: row element
    cells: (id, rowspan, colspan) structs
  */
-const fromTable = function (table: SugarElement) {
+const fromTable = (table: SugarElement) => {
   const rows = TableLookup.rows(table);
-  return Arr.map(rows, function (row) {
-    const element = row;
+  const columnGroups = TableLookup.columnGroups(table);
 
-    const parent = Traverse.parent(element);
-    const parentSection = parent.map(function (p) {
+  const tableRows = Arr.map(rows, (row) => {
+    const parent = Traverse.parent(row);
+    const parentSection = parent.map((p) => {
       const parentName = SugarNode.name(p);
-      return (parentName === 'tfoot' || parentName === 'thead' || parentName === 'tbody') ? parentName : 'tbody';
+      return Structs.isValidSection(parentName) ? parentName : 'tbody';
     }).getOr('tbody');
 
-    const cells = Arr.map(TableLookup.cells(row), function (cell) {
+    const cells = Arr.map(TableLookup.cells(row), (cell) => {
       const rowspan = getAttrValue(cell, 'rowspan', 1);
       const colspan = getAttrValue(cell, 'colspan', 1);
       return Structs.detail(cell, rowspan, colspan);
     });
 
-    return Structs.rowdata(element, cells, parentSection);
+    return Structs.rowdata(row, cells, parentSection);
   });
+
+  const tableColumnGroups = Arr.map(columnGroups, (group) => {
+    const parentOptional = Traverse.parent(group);
```

Comment:
For naming convention consistency, please use `Opt` instead of `Optional`.

Reply:
Additionally, this variable never appears to be used again, so it's just bloating the code and could be inlined.